src/components/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,177 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { IconDefinition } from "@fortawesome/free-regular-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { ThemeContext, ThemeProvider } from "../../theme";
+import { isNavbarItem } from "./Navbar.utils";
+import { isDev } from "../../utils";
+import useDotAnimation from "./hooks";
+import style from "../../variables.module.scss";
+import "./navbar.scss";
+
+const NavbarContext = React.createContext<{
+  items?: React.RefObject<HTMLLIElement>[];
+  activeId?: number;
+  setActiveId?: (id?: number) => void;
+  registerItem?: (item: React.RefObject<HTMLLIElement>) => void;
+  unRegisterItem?: (item: React.RefObject<HTMLLIElement>) => void;
+}>({});
+
+type NavbarItemProps = {
+  icon: IconDefinition;
+  title: string;
+  onSelect: ({ title, id }: { title: string; id?: number }) => void;
+  id?: number; // TODO: find out how not to expose this prop
+  iconStyles?: React.CSSProperties;
+};
+
+export function NavbarItem({
+  icon,
+  title,
+  onSelect,
+  id,
+  iconStyles,
+}: NavbarItemProps) {
+  const theme = useContext(ThemeContext);
+  const { activeId, setActiveId, registerItem, unRegisterItem } =
+    useContext(NavbarContext);
+  const itemRef = useRef<HTMLLIElement>(null);
+
+  const handleClick = () => {
+    if (id !== activeId) {
+      setActiveId?.(id);
+      onSelect({ title, id });
+    }
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    switch (event.key) {
+      case "Enter":
+      case " ":
+        event.preventDefault();
+        if (id !== activeId) {
+          setActiveId?.(id);
+          onSelect({ title, id });
+        }
+        break;
+      default:
+        break;
+    }
+  };
+
+  useEffect(() => {
+    registerItem?.(itemRef);
+    return () => unRegisterItem?.(itemRef);
+  }, [setActiveId, id, registerItem, unRegisterItem]);
+
+  const isActive = id === activeId;
+
+  return (
+    <li className="navbar-item" ref={itemRef}>
+      <button
+        onKeyDown={handleKeyDown}
```

Comment:
question/todo: Are you sure you need this?

Reply:
what?!?! 🤯 I feel Jon Snow, I really know nothing (removed since is not needed)

Reply:
Exactly! see, so many benefits from writing semantic HTML ^.^